File: aufgabe02/manschwa/SquareInRectangle.rb
puts "SquareInRectangle"
puts "============================="

width = ARGV[0].to_i
height = ARGV[1].to_i

big_square = width * height
sqrt = (Math.sqrt big_square).to_i
for e in sqrt.downto 1
  if e**2 <= big_square
    puts "#{e}"
    big_square = big_square - e**2
  end
  if big_square == e**2
    puts e
  end
end

File: aufgabe02/shoeffner/SquareInRectangle.py
# -*- coding: utf8 -*-
import sys

def SquareInRectangle(width, height):
	squares = []
	while width >= 1 and height >= 1:
		if width > height:
			squares.append(height)
			width -= height
		elif height > width:
			squares.append(width)
			height -= width
		else:
			squares.append(height)
			return squares

if __name__ == '__main__':
	print(SquareInRectangle(int(sys.argv[1]), int(sys.argv[2])))